flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherTest.java

```diff
@@ -683,6 +685,69 @@ public void testRetrieveCheckpointStats() throws Exception {
         Assertions.assertThat(resultsFuture).isCompletedWithValue(snapshot);
     }
 
+    @Test
+    public void testRetrieveCheckpointStatsOnCompletedJob() throws Exception {
+        CheckpointStatsSnapshot snapshot = getTestCheckpointStatsSnapshotWithTwoFailedCheckpoints();
+        TestingJobMasterGateway testingJobMasterGateway =
+                new TestingJobMasterGatewayBuilder().build();
+
+        dispatcher =
+                createAndStartDispatcher(
+                        heartbeatServices,
+                        haServices,
+                        new TestingJobMasterGatewayJobManagerRunnerFactory(
+                                testingJobMasterGateway));
+
+        final JobStatus expectedState = JobStatus.FAILED;
```

Comment:
Do you have issue with other completed jobs i.e. CANCELED, FINISHED? Does it make sense to test them too?

Reply:
Yep - will change to parameterized test.